File: internal/lex/tokens.go
package lex

const (
	EOF  = "EOF"
	KEYW = "KEYW"

	INT             = "INT"
	IDENT           = "IDENT"
	COM             = "//"
	COM_BLOCK_START = "/*"
	COM_BLOCK_END   = "*/"

	// KEYWORDS
	IF     = "if"
	ELSE   = "else"
	FOR    = "for"
	RETURN = "return"
	FUNC   = "func"
	VAR    = "var"
	TRUE   = "true"
	FALSE  = "false"

	// Operators
	ADD    = "+"
	ADD_E  = "+="
	INC    = "++"
	MIN    = "-"
	MIN_E  = "-="
	DEC    = "--"
	MUL    = "*"
	MUL_E  = "*="
	DIV    = "/"
	DIV_E  = "/="
	ASS    = "="
	DEF    = ":="
	COL    = ":"
	EQ     = "=="
	NOT_EQ = "!="
	BANG   = "!"
	LT     = "<"
	GT     = ">"
	LTE    = "<="
	GTE    = ">="
	ARR    = "->"

	// Delimiters
	COMMA    = ","
	LPAREN   = "("
	RPAREN   = ")"
	LBRACE   = "{"
	RBRACE   = "}"
	LBRACKET = "["
	RBRACKET = "]"
	SEMIC    = ";"
)

var Keywords = map[string]string{
	"func":   FUNC,
	"var":    VAR,
	"true":   TRUE,
	"false":  FALSE,
	"if":     IF,
	"else":   ELSE,
	"return": RETURN,
}

type Token struct {
	Type  string
	Value string
	Line  int
	Col   int
}

File: src/lex/tokens.rs
#[derive(Debug, Clone, Copy, PartialEq)]
pub enum TokenType {
    EOF,
    PARTIAL,
    SLCOM,
    MLCOM,

    IDENT,
    INT,
    CHAR,
    STR,

    EQ,    // =
    EQEQ,  // ==
    COL,   // :
    COL_E, // :=
    ADD,   // +
    ADD_E, // +=
    INC,   // ++
    MIN,   // -
    MIN_E, // -=
    ARR,   // ->
    DEC,   // --
    DIV,   // /
    MUL,   // *
    AND,   // &
    OR,    // |
    LT,    // <
    MT,    // >
    LOE,   // <=
    MOE,   // >=
    DOT,   // .
    NEQ,   // !=
    NOT,   // !

    LPAREN,
    RPAREN,
    LBRACKET,
    RBRACKET,
    LBRACE,
    RBRACE,
    COMMA,
    SEMICOL,
}

#[derive(Debug)]
pub struct Token {
    pub typ: TokenType,
    pub vs: usize,
    pub ve: usize,
    pub row: usize,
    pub col: usize,
}
